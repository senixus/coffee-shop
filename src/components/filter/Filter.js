import "./filter.scss";

const Filter = ({ handleSelect }) => {
  return (
    <div className="filter">
      <select onChange={handleSelect}>
        <option>Select</option>
        <option value="name">Name(A-Z)</option>
        <option value="name reverse">Name(Z-A)</option>
        <option value="lowest price">Lowest Price</option>
        <option value="highest price">Highest Price</option>
      </select>
    </div>
  );
};

export default Filter;
