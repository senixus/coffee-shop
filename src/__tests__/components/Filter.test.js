import React from "react";
import { shallow } from "enzyme";
import Filter from "../../components/filter/Filter";

it("expect to render Filter component", () => {
  const props = {
    handleSelect: jest.fn(),
  };
  const wrapper = shallow(<Filter handleSelect={props.handleSelect} />);
  wrapper.find("select").simulate("change");

  expect(wrapper.length).toBe(1);
  expect(props.handleSelect).toHaveBeenCalled();
  expect(props.handleSelect).toHaveBeenCalledTimes(1);
  expect(wrapper).toMatchSnapshot();
});
