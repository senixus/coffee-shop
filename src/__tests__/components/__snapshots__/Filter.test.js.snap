// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Filter component 1`] = `
<div
  className="filter"
>
  <select
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <option>
      Select
    </option>
    <option
      value="name"
    >
      Name(A-Z)
    </option>
    <option
      value="name reverse"
    >
      Name(Z-A)
    </option>
    <option
      value="lowest price"
    >
      Lowest Price
    </option>
    <option
      value="highest price"
    >
      Highest Price
    </option>
  </select>
</div>
`;
